tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,363 @@
+                # All current export is via Avro
+                uses_pandas = True
+            if uses_pandas:
+                # We're going to need to be sure to use hints that
+                # work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in datetimeformat:
+                    # Pandas is more limited in hint support than it
+                    # should be:
+                    #
+                    # https://github.com/bluelabsio/records-mover/issues/143
+                    expect_am_failure = True
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformat': datetimeformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload_column_to_string(column_name='timestamp',
+                                                        records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformat) + "\n",
+                                         # Pandas doesn't truncate fractional seconds in the
+                                         # same way other tools do.
+                                         create_sample(datetimeformat) + ".000000\n",
+                                         # An example of having an overly simplistic
+                                         # implementation of to_csv_options is that seconds get
+                                         # appended even for time formats which don't include
+                                         # it.
+                                         #
+                                         # https://github.com/bluelabsio/records-mover/issues/142
+                                         create_sample(datetimeformat) +
+                                         f":{SAMPLE_SECOND:02d}.000000\n"],
+                              f"from datetimeformat {datetimeformat} and addl_hints {addl_hints}")
+            except ModuleNotFoundError as e:
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            except NotImplementedError as e:
+                if expect_am_failure and 'AM' in str(e):
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure)
+            self.assertFalse(expect_am_failure)
+
+    def test_unload_datetimetz(self) -> None:
```

Comment:
C901 'RecordsUnloadDatetimeIntegrationTest.test_unload_datetimetz' is too complex (17)